src/Nethermind/Nethermind.Core/Transaction.cs

```diff
@@ -53,29 +56,45 @@ public class Transaction
         public bool IsMessageCall => To is not null;
 
         private Hash256? _hash;
-        public Hash256? Hash
+
+        [JsonIgnore]
+        internal bool IsHashCalculated => _hash is not null;
+        internal Hash256 CalculateHashInternal()
```

Comment:
NP: Please keep the ordering of the property. It would make it easier to review.